vision/google/cloud/vision/_rest.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""REST Client for interacting with the Google Cloud Vision API."""
+
+from google.cloud.vision.feature import Feature
+
+
+class _RESTVisionAPI(object):
+    """Vision API for interacting with the JSON/REST version of Vision
+
+    :type client: :class:`~google.cloud.core.client.Client`
+    :param client: Instance of ``Client`` object.
+    """
+
+    def __init__(self, client):
+        self._client = client
+        self._connection = client._connection
+
+    def annotate(self, image, features):
+        """Annotate an image to discover it's attributes.
+
+        :type image: str
+        :param image: A string which can be a URL, a Google Cloud Storage path,
+                      or a byte stream of the image.
+
+        :type features:  list of :class:`~google.cloud.vision.feature.Feature`
+        :param features: The type of detection that the Vision API should
+                         use to determine image attributes. Pricing is
+                         based on the number of Feature Types.
+
+                         See: https://cloud.google.com/vision/docs/pricing
+        :rtype: dict
+        :returns: List of annotations.
+        """
+        request = VisionRequest(image, features)
+
+        data = {'requests': [request.as_dict()]}
+        response = self._connection.api_request(
+            method='POST', path='/images:annotate', data=data)
+
+        return response['responses'][0]
+
+
+class VisionRequest(object):
```

Comment:
Are there future uses of this class coming? Seems to me that it should just be a function